src/renderers/native/ReactNativeAttributePayload.js

```diff
@@ -314,6 +318,9 @@ function diffProperties(
   for (var propKey in nextProps) {
     attributeConfig = validAttributes[propKey];
     if (!attributeConfig) {
+      if (__DEV__ && enableStrictAttributeValidation === true && propKey !== 'children') {
```

Comment:
This entire block should be wrapped in `if` block that just checks `__DEV__`, so that the code can be removed completely in a production build

```js
if (__DEV__) {}